src/Core/System/Snippet/SnippetFileHandler.php

```diff
@@ -7,28 +7,49 @@
 use Shopware\Storefront\Storefront;
 use Symfony\Component\Finder\Finder;
 
+/**
+ * @phpstan-type Snippets array<string, string|mixed>
+ */
 #[Package('discovery')]
 class SnippetFileHandler
 {
+    /**
+     * @return Snippets
+     */
     public function openJsonFile(string $path): array
     {
-        $json = json_decode(file_get_contents($path), true);
+        $fileContents = file_get_contents($path);
+
+        if ($fileContents === false) {
+            throw SnippetException::jsonNotFound();
+        }
+
+        $json = json_decode($fileContents, true, 512, \JSON_THROW_ON_ERROR);
 
         $jsonError = json_last_error();
```

Comment:
as you added the `JSON_THROW_ON_ERROR` flag to `json_decode` this would never be called :slightly_smiling_face: instead you should catch the \JsonException